org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/TypeUtils.java

```diff
@@ -0,0 +1,222 @@
+    static final char GENERIC_END = '>';
+    static final char COLON = ':';
+
+    static Map<Character, String> primitiveTypeMap = new HashMap<Character, String>();
+
+    static {
+        primitiveTypeMap.put(BYTE, "byte");
+        primitiveTypeMap.put(CHAR, "char");
+        primitiveTypeMap.put(FLOAT, "float");
+        primitiveTypeMap.put(DOUBLE, "double");
+        primitiveTypeMap.put(INT, "int");
+        primitiveTypeMap.put(LONG, "long");
+        primitiveTypeMap.put(SHORT, "short");
+        primitiveTypeMap.put(VOID, "void");
+        primitiveTypeMap.put(BOOLEAN, "boolean");
+    }
+
+    /**
+     * Returns the short name and signature of the JDI type, eg: <code>HashMap&lt;K,V&gt;</code> for
+     * <code>java.util.HashMap</code>.
+     *
+     * @param type
+     *            the JDI type for evaluate
+     * @return the display name
+     */
+    public static String getDisplayName(Type type) {
+        return getDisplayName(type, false);
+    }
+
+    /**
+     * Returns the name and signature of the JDI type, eg: <code>java.util.HashMap&lt;K,V&gt;</code>
+     * for <code>java.util.HashMap</code>.
+     *
+     * @param type
+     *            the JDI type for evaluate
+     * @param showQualified
+     *            whether or not to show the fully qualified class name
+     * @return the display name
+     */
+    public static String getDisplayName(Type type, boolean showQualified) {
+        if (type == null) {
+            throw new UnsupportedOperationException("Array type is not supported.");
+        }
+        String signature = type.signature();
+        if (signature.charAt(0) == ARRAY) {
+            throw new UnsupportedOperationException("Array type is not supported.");
+        }
+        if (isObjectTag(signature.charAt(0))) {
+            return getDisplayNameForReferenceType((ReferenceType) type, showQualified);
+        } else {
+            String typeName = getTypeName(signature);
+            return showQualified ? typeName : trimTypeName(typeName);
+        }
+    }
+
+
+    /**
+     * Find the reference type from JVM.
+     *
+     * @param vm
+     *            the JVM.
+     * @param expectedClassLoader
+     *            the class loader of the class(because JVM many have two or more classes
+     *            with same full qualified name)
+     * @param typeName
+     *            the full qualified class name
+     * @return the reference type
+     */
+    public static ReferenceType resolveReferenceType(VirtualMachine vm,
+            ClassLoaderReference expectedClassLoader, String typeName) {
+        List<?> list = vm.classesByName(typeName);
+        Iterator<?> iter = list.iterator();
+        while (iter.hasNext()) {
+            ReferenceType componentType = (ReferenceType) iter.next();
+            ClassLoaderReference cl = componentType.classLoader();
+            if ((cl == null) ? (expectedClassLoader == null) : (cl.equals(expectedClassLoader))) {
```

Comment:
if user didn't pass expectedClassLoader, i suggest to return the first found ReferenceType. No need to compare ClassLoader.